### PR TITLE
Update securing_applications guide for latest adapter changes (community)

### LINKED ISSUE #20994: Update securing_applications guide for latest adapter changes (community)
### Description

There are few pieces of our "Securing applications and services guide", which are not up-to-date based on latest adapter changes.

### Discussion

_No response_

### Motivation

_No response_

### Details

_No response_

### PR DESCRIPTION
**Pull request by mposolda**
closes #20994

- Added note that our OIDC java adapters are deprecated and applicable only to JEE
- Removed note from our OIDC JBoss adapter docs that "We test and support latest Wildfly version" as this is not accurate anymore. Instead added that people are encouraged to use elytron-oidc-client
- Some updates to SAML docs for Jakarta
- Some deprecation of SAML adapters (Jetty, Tomcat) and note that we don't support mod_auth_mellon.

### CODE REVIEW

**Review by pedroigor — COMMENTED** [review]
@mposolda Thanks.

**Comment by mposolda** [comment]
@pedroigor Applied your suggestion. Thanks!

File: docs/documentation/securing_apps/topics/oidc/java/jboss-adapter.adoc [hunk 1]
```diff
@@ -17,6 +17,11 @@ You can install this adapter from a ZIP file or from an RPM.
 endif::[]
 
 ifeval::[{project_community}==true]
+
+WARNING: We do not regularly test and maintain WildFly/EAP adapters. They may work only with WildFly version of 23 or earlier or with JBoss EAP 7. Also note
```
Comment (andymunro) [inline docs/documentation/securing_apps/topics/oidc/java/jboss-adapter.adoc:21]:
```suggestion
WARNING: We do not regularly test and maintain WildFly/EAP adapters. They may work only with WildFly version 23 or earlier or with JBoss EAP 7. Also note
```

File: docs/documentation/securing_apps/topics/oidc/java/jboss-adapter.adoc [hunk 2]
```diff
@@ -17,6 +17,11 @@ You can install this adapter from a ZIP file or from an RPM.
...
+that they may work only with JEE applications, but not with Jakarta applications. It is recommended to switch to Elytron OIDC Java adapter to
+secure your applications. It has very similar configuration to Keycloak Java adapters and migrating your applications to it should be smooth.
```
Comment (andymunro) [inline docs/documentation/securing_apps/topics/oidc/java/jboss-adapter.adoc:23]:
```suggestion
that they may work only with JEE applications, but not with Jakarta applications. We recommend that you switch to Elytron OIDC Java adapter to
secure your applications. This adapter has a similar configuration to Keycloak Java adapters and migrating your applications to it should be smooth.
```

File: docs/documentation/securing_apps/topics/saml/java/jetty-adapter.adoc [hunk 3]
```diff
@@ -2,6 +2,8 @@
 
 ==== Jetty SAML adapters
 
+WARNING: {project_name} Jetty SAML adapter is deprecated. It is recommended to use other client adapter if possible.
```
Comment (andymunro) [inline docs/documentation/securing_apps/topics/saml/java/jetty-adapter.adoc:5]:
```suggestion
WARNING: The {project_name} Jetty SAML adapter is deprecated. We recommend that you use another client adapter if possible.
```

File: docs/documentation/securing_apps/topics/saml/java/tomcat-adapter.adoc [hunk 4]
```diff
@@ -2,6 +2,8 @@
 
 ==== Tomcat SAML adapters
 
+WARNING: {project_name} Tomcat SAML adapter is deprecated. It is recommended to use other client adapter if possible.
```
Comment (andymunro) [inline docs/documentation/securing_apps/topics/saml/java/tomcat-adapter.adoc:5]:
```suggestion
WARNING: The {project_name} Tomcat SAML adapter is deprecated. We recommend that you use another client adapter if possible.
```

File: docs/documentation/securing_apps/topics/saml/mod-auth-mellon.adoc [hunk 5]
```diff
@@ -2,6 +2,9 @@
 
 === mod_auth_mellon Apache HTTPD Module
 
+WARNING: {project_name} does not provide any official support to mod_auth_mellon. The instructions below are best-effort and may not be up-to-date.
+It is recommended to stick to official mod_auth_mellon documentation for more details.
```
Comment (andymunro) [inline docs/documentation/securing_apps/topics/saml/mod-auth-mellon.adoc:6]:
```suggestion
WARNING: {project_name} does not provide any official support to mod_auth_mellon. The instructions below are best-effort and may not be up-to-date.
We recommend that you stick to official mod_auth_mellon documentation for more details.
```

**Review by andymunro — CHANGES_REQUESTED** [review]
Looks good, @mposolda. A few changes to suggest.

**Comment by mposolda** [comment]
@andymunro Applied your changes. Thanks!

**Pull request merged**